### PR TITLE
integration of custom alerting

### PR DESCRIPTION
**Pull request by junotx**

### CODE REVIEW

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

File: roles/ks-monitor/files/prometheus/thanosruler/thanosruler.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+              operator: In
+              values:
+              - k8s
+          namespaces:
+          - kubesphere-monitoring-system
+          topologyKey: kubernetes.io/hostname
+        weight: 100
+  alertmanagersUrl:
+  - dnssrv+http://alertmanager-operated.kubesphere-monitoring-system.svc:9093
+  image: quay.io/thanos/thanos:v0.18.0
```
Comment (benjaminhuo) [inline roles/ks-monitor/files/prometheus/thanosruler/thanosruler.yaml:23]: It better to be the image from the dockerhub repo here too: `thanosio/thanos`

Comment (pixiake) [reply roles/ks-monitor/files/prometheus/thanosruler/thanosruler.yaml]: Image name is better as a parameter in https://github.com/kubesphere/ks-installer/blob/master/roles/download/defaults/main.yml

**Pull request merged**